server/Student.ts

```diff
@@ -90,6 +93,24 @@ class Student extends EventEmitter {
         }
     }
 
+    handleSuicide() {
+        const response: Payload = {
+            event: "you_dead"
+        };
+        this.wsc?.send(JSON.stringify(response));
+        if(!this.wsc?.isClosed){
+            this.wsc?.close(1000, "Student requested shutdown");
+        }
+        this.handleRodo();
+        this.lecture.studentList.deleteStudent(this.id);
+    }
+
+    handleRodo(){
+        this.name = "";
+        this.surname = "";
+        this.nick = "";
+    }
```

Comment:
What purpose is this function if a student gets deleted?

Reply:
to be sure that personal data is deleted, so I overwrite it

Reply:
GDPR not RODO
https://en.wikipedia.org/wiki/General_Data_Protection_Regulation

And i dont think we need such function we don't store users in database.